### ISSUE TITLE
[Feature] RandomHorizontalFlipOp / RandomVerticalFlipOp

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#31]
## Summary

Add `RandomHorizontalFlipOp` and `RandomVerticalFlipOp` data augmentation operations.

## Motivation

Random flipping is one of the most fundamental and universally used data augmentation techniques in computer vision. Virtually every image training pipeline uses horizontal/vertical flips. This is a critical gap for a preprocessing library.

## Reference Implementations

- **PyTorch**: `torchvision.transforms.RandomHorizontalFlip(p=0.5)`, `torchvision.transforms.RandomVerticalFlip(p=0.5)` — [PyTorch docs](https://pytorch.org/vision/stable/generated/torchvision.transforms.RandomHorizontalFlip.html)
- **TensorFlow**: `tf.image.random_flip_left_right(image)`, `tf.image.random_flip_up_down(image)` — [TF docs](https://www.tensorflow.org/api_docs/python/tf/image/random_flip_left_right)
- **tflite_flutter_helper**: No direct flip op, but commonly needed

## Verification

- ✅ Feature exists in PyTorch (torchvision.transforms.RandomHorizontalFlip) — VERIFIED
- ✅ Feature exists in TensorFlow (tf.image.random_flip_left_right) — VERIFIED
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for flip, horizontal, vertical, mirror)

## Proposed API

```dart
// Horizontal flip with 50% probability (default)
final flipH = RandomHorizontalFlipOp(probability: 0.5);

// Vertical flip with 30% probability
final flipV = RandomVerticalFlipOp(probability: 0.3);

// Deterministic flips (always apply)
final flipH = HorizontalFlipOp();
final flipV = VerticalFlipOp();
```

## Implementation Notes

- Should support 3D `[C, H, W]` and 4D `[N, C, H, W]` tensors
- Extend `TransformOp` base class
- Can be implemented as zero-copy stride manipulation (negative stride along width/height axis)
- Add `OperationCapabilities` with:
  - `pytorchEquivalent: 'torchvision.transforms.RandomHorizontalFlip'`
  - `onnxOpType: null` (no direct ONNX equivalent)
- Add optional seed parameter for reproducibility

## ROI Assessment

**Very High** — This is one of the most commonly used augmentations in any vision pipeline. Implementation is straightforward via stride manipulation. High impact, low effort.
